siliconcompiler/core.py

```diff
@@ -4211,6 +4211,18 @@ def _local_process(self, flow, status, steplist, indexlist):
         for (step, index) in flowgraph_nodes:
             # Setting up tool is optional
             self._setup_node(step, index)
+            # Env vars are necessary as test_multiple_tools.py requires it for its version check
+            self._set_env_vars(step, index)
+            tool, task = self._get_tool_task(step, index, flow)
+            # Icarus compiles its executable during the run so we can't check its version
+            if (tool, task) != ('execute', 'exec_input'):
```

Comment:
I think `check_tool_version` should handle this by checking if there is a version to test

Reply:
Currently it just fails and says the executable could not be found.
I can move this check into `check_tool_version` though.

Reply:
I don't think this check is correct, we should find a better method then exclude a single node

Reply:
I'll look what verilator does differently in its tool setup since that should also produce an executable dynamically.

Reply:
Fixed it in icarus by using the dummy executable `true` instead of a non existing executable.